services/gossip/adapter/tcp/direct_transport.go

```diff
@@ -112,31 +111,26 @@ func newDirectTransportServer() *lockableTransportServer {
 	}
 }
 
-func newLockableOutgoingQueues() *lockableOutgoingQueues {
-	return &lockableOutgoingQueues{
-		peers: make(map[string]*transportQueue),
+func newLockableClientConnections() *lockableClientConnections {
+	return &lockableClientConnections{
+		peers: make(map[string]*clientConnection),
 	}
 }
 
 // note that bgCtx MUST be a long-running background context - if it's a short lived context, the new connection will die as soon as
 // the context is done
 func (t *DirectTransport) connectForever(bgCtx context.Context, peerNodeAddress string, peer config.GossipPeer) {
-	t.outgoingQueues.Lock()
-	defer t.outgoingQueues.Unlock()
+	t.clientConnections.Lock()
+	defer t.clientConnections.Unlock()
 
 	if peerNodeAddress != t.config.NodeAddress().KeyForMap() {
 
-		newQueue := NewTransportQueue(SEND_QUEUE_MAX_BYTES, SEND_QUEUE_MAX_MESSAGES, t.metricRegistry, peerNodeAddress)
+		client := newClientConnection(peer.HexAddress()[:6], peer, t.logger, t.metricRegistry, t.metrics, t.config)
```

Comment:
why do you enforce the length of HexAddress here?

Reply:
I do not enforce the length - I use only the first 6 chars